app/Models/Device.php

```diff
@@ -330,7 +331,19 @@ public function setLocation($location_text)
 
         $this->location_id = null;
         if ($location_text) {
-            $location = Location::firstOrCreate(['location' => $location_text]);
+            $location_data = ['location' => $location_text];
+
+            if (\LibreNMS\Config::get('parse_dns_location_record')) {
```

Comment:
I believe this is in the wrong place, for once, it will only run if `$location_text` is set.
Maybe it should be in;
https://github.com/librenms/librenms/blob/53bc78c9d93679babeb67d8b1a14a9578ccb6572/LibreNMS/Modules/OS.php#L103

@murrant ?

Reply:
@Jellyfrog good point,
@murrant some better ideas for positioning it?

Reply:
or are you fine with @Jellyfrog 's thought?

Reply:
Jellyfrog is correct.  The model function should be just to set the actual location field and not do any extra processing.

Reply:
moved :-)